README.md

```diff
@@ -215,19 +196,73 @@ We got these results:
 
 ![histogram](benchmarks/histogram/distribution.png)
 
-The histogram represents the number of files for which spent time in language
-detection was in the range of the time interval indicated in the x axis.
+The histogram shows the number of files detected (y-axis) per time interval bucket (x-axis). As one can see, most of the files were detected faster by enry.
 
-So you can see that most of the files were detected quicker in enry.
+We found few cases where enry turns slower than linguist due to
+Go regexp engine being slower than Ruby's, based on [oniguruma](https://github.com/kkos/oniguruma) library, written in C.
 
-We found some few cases where enry turns slower than linguist. This is due to
-Golang's regexp engine being slower than Ruby's, which uses the [oniguruma](https://github.com/kkos/oniguruma) library, written in C.
+See (instructions)[#faster-regexp-engine-optional] for running enry with oniguruma.
 
-You can find scripts and additional information (like software and hardware used
-and benchmarks' results per sample file) in [*benchmarks*](https://github.com/src-d/enry/blob/master/benchmarks) directory.
 
+Why Enry?
+------------
+
+In the movie [My Fair Lady](https://en.wikipedia.org/wiki/My_Fair_Lady), [Professor Henry Higgins](http://www.imdb.com/character/ch0011719/?ref_=tt_cl_t2) is one of the main characters. Henry is a linguist and at the very beginning of the movie enjoys guessing the origin of people based on their accent.
+
+`Enry Iggins` is how [Eliza Doolittle](http://www.imdb.com/character/ch0011720/?ref_=tt_cl_t1), [pronounces](https://www.youtube.com/watch?v=pwNKyTktDIE) the name of the Professor during the first half of the movie.
+
+
+## Development
+
+To build enry's CLI run:
+
+    make build
+
+this will generate a binary in the project's root directory called `enry`.
+
+To run the tests:
+
+    make test
 
-### Benchmark Dependencies
+
+### Sync with github/linguist upstream
+
+*enry* re-uses parts of the original [github/linguist](https://github.com/github/linguist) to generate internal data structures.
+In order to update to the latest release of linguist do:
+
+    git clone https://github.com/github/linguist.git .linguist
+
+    # put the new release's commit sha in the generator_test.go (to re-generate .gold test fixtures)
```

Comment:
Before this should be noted that you should also `cd .linguist && git checkout <latest_relase_sha>`